### PR TITLE
[Fix] fix bug of show_results in single_gpu_test

### PR DESCRIPTION
**Pull request by cuhk-hbsun**

### CODE REVIEW

**Comment by cuhk-hbsun** [comment]
fix #656

File: mmocr/apis/test.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+import mmcv
+import numpy as np
+import torch
+from mmcv.image import tensor2imgs
+from mmcv.parallel import DataContainer
+from mmdet.core import encode_mask_results
+
+
+def tensor2grayimgs(tensor, mean=(127), std=(127), **kwargs):
```
Comment (gaotongxiao) [inline mmocr/apis/test.py:12]: Unit test ref: https://github.com/open-mmlab/mmcv/blob/c47c9196d067a0900b7b8987a8e82768edab2fff/tests/test_image/test_image_misc.py#L15

File: mmocr/apis/test.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                img_metas = data['img_metas'].data[0]
+                must_keys = ['img_norm_cfg', 'ori_filename']
+                for key in must_keys:
+                    if key not in img_metas[0]:
+                        raise KeyError(
+                            f'Please add {key} to the "meta_keys" in config.')
+                if len(img_metas[0]['img_norm_cfg']['mean']) != 3:
+                    raise Exception(
+                        'Only model with visual is supported here to show '
+                        'results. Please use tools/kie_test_imgs.py instead.')
```
Comment (gaotongxiao) [inline mmocr/apis/test.py:119]: It seems the code in that script is not too complex and can be adapted here.

File: mmocr/apis/test.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    if img_tensor.size(1) == 3:
+                        h, w, _ = img_meta['img_shape']
+                        img_show = img[:h, :w, :]
+                        ori_h, ori_w = img_meta['ori_shape'][:-1]
+                    elif img_tensor.size(1) == 1:
+                        h, w = img_meta['img_shape']
+                        img_show = img[:h, :w]
+                        ori_h, ori_w = img_meta['ori_shape']
+
+                    img_show = mmcv.imresize(img_show, (ori_w, ori_h))
```
Comment (Harold-lkk) [inline mmocr/apis/test.py:140]:
```suggestion
                    img_show = img[:img_meta['img_shape'][0], img_meta['img_shape'][1]]
                    img_show = mmcv.imresize(img_show, 
                                            (img_meta['ori_shape'[1],
                                             img_meta['ori_shape'][0]))
```

**Comment by Harold-lkk** [comment]
link to https://github.com/open-mmlab/mmcv/pull/1595

**Review by Harold-lkk — APPROVED** [review]

**Pull request merged**